RDK-59589, RDKEMW-9377: AAMP Federated Release for Sprint 2510 - Tag 2.10.0

psiva01 opened the pull request.

pstroffolino committed:
VPLAY-10974 l1test to ensure no compilation issues with aamp plugins

VPLAY-10974 l1test to ensure no compilation issues with aamp plugins

Reason for Change: ensure plugins (i.e. aampcdmidecryptor) compile, and allow their code to be processed by Xcode static code analysis

Test Guidance: run l1 tests on OSX

Risk: Low
---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

supriyadappu committed:
VPLAY-10980: Aamp tune Version is displayed 7.07 Instead of 7.08 (#477)

VPLAY-10980: Aamp tune Version is displayed 7.07 Instead of 7.08 

Reason for change: Changed AAMP version to 7.08
Test Procedure: Refer Jira
Risks: No

pstroffolino committed:
VPLAY-11009 aamp cli logs seekablerange as "seekablerangen/a" when not applicable (#479)

VPLAY-11009 aamp cli logs seekablerange as "seekablerangen/a" when not applicable

Reason for Change: seekable range logged in confusing manner when tsb not available
Test Guidance: play linear stream in aampcli without use of local TSB and check AAMP_EVENT_PROGRESS logs
Risk: none

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-11027: [Doc] decode time from decryption to first frame (#482)

VPLAY-11027: [Doc] decode time from decryption to first frame

Reason for change: Update documentation and add the new gstDecodeTime field to TuneTimeMetrics

Test Procedure: Just a documentation update
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

psiva01 committed:
VPLAY-10983: Bring back VPLAY-9850 changes with VPLAY-10930 crash (#490)

Revert "Revert "VPLAY-9850 Deprecate need for GrowableBuffer::AppendNulTerminator" for VPLAY-10930 (#472)"

Reason for change: Added thread safety to AampGrowableBuffer::Free(),
 ReserveBytes() and AppendBytes().
Test Procedure: Refer Jira
Risks: Low

---------

Signed-off-by: psiva01 <sivasubramanian.patchaiperumal@ltts.com>

jfagunde committed:
VPLAY-11040 Reduce logging from PushNextTsbFragment() (#489)

VPLAY-11040 Reduce logging from PushNextTsbFragment()

Reason for Change: Log output from PushNextTsbFragment() during trick modes is very noisy

Summary of Changes:
* Do not call PushNextTsbFragment() for tracks that are not enabled
* During FF "Failed to read next fragment" is expected so moved to TRACE
* Improve logging and comments
* Verify PushNextTsbFragment is not called if track is disabled in L1 test
* Fix comments and rename test class member variables
* Clarify 'live edge' in comment and log line
* Convert noisy INFO to TRACE and fix L1 tests

Test Procedure: Check log lines from PushNextTsbFragment() when using trick modes

Risk: Low

---------

Signed-off-by: Jose Fagundez <jfagunde@synamedia.com>

Reshma-JO07 committed:
VPLAY-10777: Index 0 does not maps to lowest profile (#457)

VPLAY-10777: Index 0 does not maps to lowest profile

Reason for change: Added an api in ABRManager to get the lowest bandwidth index
Test Procedure: Updated in ticket
Risks: Medium

Signed-off-by: Reshma-JO07 <sreshmaraphaelk@gmail.com>

pstroffolino committed:
VPLAY-11128 GrowableBuffer test crashing on OSX (#499)

VPLAY-11128 GrowableBuffer test crashing on OSX

Reason for Change: avoid crash when running these tests on OSX

Test Guidance: problematic tests skipped on OSX

Risk: Low (test-only change)

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

* Reason for Change: improve comments; make GTEST_SKIP platform dependent

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10763 Gstreamer Test Harness problem trouble playing url contai… (#501)

VPLAY-10763 Gstreamer Test Harness problem trouble playing url containing "@" character

Reason for Change: avoid ambiguous use of @ in locators; used to designate range
also commented out noisy enough_data logging

Test Guidance: test with locators like file:///Users/@email.com/Downloads/multiIframeTracks/manifest.mpd

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10777 panic mode heuristic change

VPLAY-10777 panic mode heuristic change

Reason for Change: relax panic mode check to happen with buffering <2, rather than only with buffering=0

Test Guidance: see ticket

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

jfagunde committed:
VPLAY-10931 Mute SubTec when video is muted (#497)

VPLAY-10931 Mute SubTec when video is muted

Reason for change: Refactoring to improve efficiency and readability

Summary of changes:
* Call SubTec with (video_muted || subtitles_muted)
* Move SetCCStatus logic from PlayerInstanceAAMP to PrivateInstanceAAMP
* Delete CacheAndApplySubtitleMute() as it is no longer required
* Update existing L1 tests related to subtitles
* Improve logging and code to address review comments
* Refactor SetCCStatus() code to make it clearer

Risks: Low

Test Procedure: Turn subtitles on and check regressions when
enabling/disabling video or subtitles (e.g. due to a PIN request)

pstroffolino committed:
VPLAY-11133  after Xcode upgrade, facing error with "minimum deployment target" (#506)

VPLAY-11133  after Xcode upgrade, facing error with "minimum deployment target"

Reason for Change: deal with xcode update; adde explicit CMAKE_OSX_DEPLOYMENT_TARGET

Test Guidance: delete build directory, run install-aamp.sh, and confirm xcode project builds/runs

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

jameslofthouse-RED committed:
VPLAY-11152 crash during first playback after coming out of deepsleep (#505)

VPLAY-11152 crash during first playback after coming out of deepsleep 

Reason for change: Initialise AampDRMLicenseManager class members
Test Procedure: see ticket
Risks: Low
Signed-off-by: James Lofthouse <james_lofthouse@comcast.com>

pstroffolino committed:
VPLAY-11150 use of memory allocated with size 0 (#504)

VPLAY-11150 use of memory allocated with size 0

Reason for Change: minor refactoring to avoid a pedantic compiler warning
Test Guidance: warning no longer seen with xcode Analyze Code
Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

psiva01 committed:
Revert "VPLAY-10983: Bring back VPLAY-9850 changes with VPLAY-10930 crash (#490)" (#508)

This reverts commit c12d08e61ae95141e76c8603d5bb852561218abc.

dp0000 committed:
VPLAY-11219:[VIPA-WATERMARKING] Watermarking not rendering on top of … (#515)

VPLAY-11219:[VIPA-WATERMARKING] Watermarking not rendering on top of VIPA App with Firebolt
Reason for change: Watermarking not rendered on top of vipa widget
Test Procedure: Refer Jira
Risks: Low

---------

Signed-off-by: dn808 <deepikasri_n@comcast.com>

pstroffolino committed:
VPLAY-11211 subtec workflow (oob caption download and injection) should happen in simulator even if subtec not running (#512)

VPLAY-11211 subtec workflow (oob caption download and injection) should happen in simulator even if subtec not running

Reason for Change: allow subtitle download and fake-injection to happen on simulator even if no subtec running

Test Guidance: oob subtitle downloads happening even when subtec not running

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

* Update PacketSender.cpp

---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

srikanthreddybijjam-comcast committed:
VPLAY-9990: Fix for COPY_INSTEAD_OF_MOVE Coverity Issue (#445)

VPLAY-9990: Fix for COPY_INSTEAD_OF_MOVE Coverity Issue (#445)

Reason for change :Resolved Coverity Issue caused by(COPY_INSTEAD_OF_MOVE)
Test Procedure: Refer jira ticket VPLAY-9990
Priority: P1

Signed-off-by: srikanthreddybijjam-comcast <srikanthreddybijjam.2000@gmail.com>

jfagunde committed:
VPLAY-11125 [AAMP-TSB] Fragments not skipped when ff to live at high rate (#502)

VPLAY-11125 [AAMP-TSB] Fragments not skipped when ff to live at high rate

Reason for Change: When ff to live, not inject fragments if the next one is not available

* Handle the BoS with skipped fragment scenario
* Add L1 test to verify SkipFragment when rw to BoS
* Stop skipping fragments after reaching the live play position
* Move rate checks inside SkipFragment() and other review comments
* Split SkipFragment() to simplify code
* Always print NavigateToNextFragment() INFO when reaching BoS

Test Guidance: see ticket

pstroffolino committed:
Revert "VPLAY-11211 subtec workflow (oob caption download and injection) shou…" (#529)

This reverts commit 8876e4a68043d0392a57423b39f10c974457a3d0.

pstroffolino committed:
VPLAY-9990 avoid double std::move use with drmHelperToUse (#534)

VPLAY-9990 avoid double std::move use with drmHelperToUse

Reason for Change: avoid xcode-flagged warning from static code analysis related to back-to-back std::move

Test Guidance: no more warning in xcode

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

* Update fragmentcollector_hls.cpp

std::move only on 2nd use of drmHelperToUse

---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

varatharajan568 committed:
VPLAY-11189 Map relevant CP Thunder Plugin Error Codes in AAMP

VPLAY-11189 Map relevant CP Thunder Plugin Error Codes in AAMP

Reason for Change: Mapped the relevant CP Thunder Plugin Error Codes in AAMP
Test Guidance: Need to check the CP failure scenario
Risk: Low

* Handled the mapping CPS to AAMP error code during problematic scenario alone
* Provided the watermark failure reason code to application

---------

Signed-off-by: Varatharajan_Narayanan <Varatharajan_Narayanan@comcast.com>

Vinish100 committed:
VPLAY-11140: Encrypted playback support with mp4demux (#536)

VPLAY-11140: Encrypted playback support with mp4demux (#536)

Reason for change: Set encrypted caps if stream is encrypted.
Also add the protection meta data to buffer for decryptor.
Add support for encrypted LLD streams with mp4demux.
Test Procedure: Set 'useMp4Demux=true' and ensure encrypted content plays
Risks: Low
Priority: P1

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

varatharajan568 committed:
RDK-56050 Merging of OCDM, gst-plugins-rdk into Library

RDK-56050 Merging of OCDM, gst-plugins-rdk into Library

Reason for change: All the middleware components(gstreamerpiplelinecalls,subtec,CC,drm,OCDM,gst-plugins) are brought inside playergstinterface library.  All the L1 test cases related to playergstinterface and playerfbinterface and brought inside this Middleware folder removing their linkage from AAMP

Test Procedure: Refer Ticket

Risks: Low

Priority: P1

Change-Id: If0a31d066478806bae29f09ab829b1108d7856ba
Signed-off-by: Deepikasri N <Deepikasri_N@Comcast.Com

pstroffolino committed:
RDK-56050 patche to build script to make them work on OSX (#540)

RDK-56050 patche to build script to make them work on OSX (#540)

Reason for Change: fix osx build issue after recent middleware migration

Test Guidance: below should work:
cd middleware
./install-middleware.sh
cd test/utests/
./run.sh

Risk: Low (already broken)

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-11271: AAMP build issue in MACOS 15.6

Reason for change: ASIO version mismatch is happening
in MACOS 15.6, extend the support for building asio 1.18.2
to MACOS 15.5 onwards.
Test Procedure: Make sure AAMP builds properly in MACOS 15.6
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

Vinish100 committed:
VPLAY-11269: Fix mem leak in AampProfiler (#541)

VPLAY-11269: Fix mem leak in AampProfiler (#541)

Reason for change: Release the memory allocated to convert
json to string. Also handle mLldLowBuffObject properly
Test Procedure: Make sure no mem leaks observed on
tune tests. Refer ticket for some cases
Risks: Low
Priority: P0

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

anshephe committed:
VPLAY-11042: [VIPA][AAMP TSB] No ad progress reports when resuming normal rate 1x playback after seeking or trick-playing into an ad break (#509)

VPLAY-11042: [VIPA][AAMP TSB] No ad progress reports when resuming normal rate 1x playback after seeking or trick-playing into an ad break (#509)

Reason for Change:
* If seek (or resume from trickplay) within ad advert reservation period, raise the RESERVATION_START event
* Suppress PLACEMENT_END, and RESERVATION_END if not had the START events.
* Coverity changes
* Change to the sorting logic for ad events at the same position
* Correction to the AD event sorting
* Coverity fixes

Test Guidance: refer ticket

Risk: Low
---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

DomSyna committed:
VPLAY-11187 GetAvailableTextTracks() mark in-band CC tracks as available  (#519)

VPLAY-11187 GetAvailableTextTracks() mark in-band CC tracks as available  (#519)

Reason for Change: foundational work and l1 tests; In band tracks are now marked as available when using AAMP Local TSB

Test Guidance: basic regression only - no functional change expected; will be tested as part of inband caption feature delivery

Risk: Low
---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

paulpandian25 committed:
Revert "RDK-56050 patche to build script to make them work on OSX (#540)" (#544)

This reverts commit d884361acb068185e5bc25bb2aa69c2e87749dad.

Revert "RDK-56050 Merging of OCDM, gst-plugins-rdk into Library"

This reverts commit 04b5b93bb87b6255448d6aa1b10a0e94f11ecb4b.

Co-authored-by: Paulpandian Mariappan <paulpandian_mariappan@cable.comcast.com>

pstroffolino committed:
VPLAY-11269 FakeAampProfiler patch (#546)

Reason for Change: was missing pointer initialization, causing many tests to crash on OSX

Test Guidance: L1 tests passing on both Ubuntu and OSX

Risk: None

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10930 [AAMP][Xumo] - Crash is Observed with finger print "43705027" while changing linear Channel (#513)

VPLAY-10930 [AAMP][Xumo] - Crash is Observed with finger print "43705027" while changing linear Channel

* Restored refactoring with Revert "Revert "VPLAY-10983: Bring back VPLAY-9850 changes with VPLAY-10930 c…"

This reverts commit 2f0d7fd419b17897daece8e869bf0eee24320b41.

* Updates fragmentcollector_hls.cpp with strategically added padding to avoid Rialto crash

---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
Revert "Revert "RDK-56050 Merging of OCDM & patche to build script in OSX" (#547)

Revert "Revert "RDK-56050 patches to build script to make them work on OSX (#5…"

Reason for Change: Restore RDK-56050 changes.  Fix OSX simulator build regression on newer Mac.  Fix misc coverity issues.

Test Guidance: see ticket - no functional change expected - pure regression

Risk: Low

This reverts commit f753e1cb76b102573b6fd4180ee5f0a70d576ac0.

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-11287 deprecate x1-specific "classic tune" logging and eAAMPConfig_XRESupportedTune (#545)

VPLAY-11287 deprecate x1-specific "classic tune" logging and eAAMPConfig_XRESupportedTune

Reason for Change: deprecate some no-longer needed code (legacy x1-specific logging)

Test Guidance: no regression with retained IP_AAMP_TUNETIME logging

Risk: Low
---------

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

psiva01 committed:
VPLAY-11062: Crash observed with Fingerprint "42808043" (#549)

VPLAY-11062: Crash observed with Fingerprint "42808043" (#549)

Reason for change: Set EventManager State to RELEASED in
PrivateInstanceAAMP::Stop() to avoid event processing.
Test Procedure: Refer Jira
Risks: No

Signed-off-by: psiva01 <sivasubramanian.patchaiperumal@ltts.com>

varshnie committed:
VPLAY-10567:[AAMP] Profile playback stop operation (#557)

VPLAY-10567:[AAMP] Profile playback stop operation

Reason for change: include "stopTime" (for previous time) in IP_TIMETIME metrics/logging
Test Procedure: ticket
Priority: P1
Risk: Low
---------

Signed-off-by: varshnie <varshniblue14@gmail.com>

nu641001 committed:
VPLAY-10909: Get PTO honoured and PTS restamp working at the same time (#459)

VPLAY-10909: Get PTO honoured and PTS restamp working at the same time (#459)

Reason for change: Flush the pipeline if there is a presentation time offset.
For other streams, restamping works without flush

Risks: Low

Test Procedure: Test with DASH (especially Charter) streams

Priority: P1

Signed-off-by: nu641001 <nu641001@gmail.com>

jfagunde committed:
VPLAY-11356 Verify SetCCStatus() in L1 tests (#564)

* VPLAY-11356 Verify SetCCStatus() in L1 tests

* Verify SetCCStatus() call when StreamAbstraction object is created

* VPLAY-11356 Update the copyright date for MockPlayerCCManager.h

* Add blank lines to MockPlayerCCManager.h

---------

Co-authored-by: pstroffolino <Philip_Stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-11062 revert (#567)

Reason for Change: this broke l2 tests 1006 and 1018; player state not changing after stop

Test Guidance: l2 tests again passing

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-10777 restoration - broken by SocSep changes RDK-56050 (#568)

VPLAY-10777 restoration - broken by SocSep changes RDK-56050

Reason for Change: RDK-56050 inadvertantly stomped on some recent core aamp fixes

Test Guidance: retest L2 4050 and 2036 and VPLAY-10777

Risk: Low

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

vinodkadungoth committed:
VPLAY-11265:AAMP Playback Failure During Dynamic Ad Insertion in Paused Live DAI Stream (#562)

VPLAY-11265: Disabled downloads for ePAUSED_BEHAVIOR_LIVE_DEFER

Reason for change:
Test Procedure: Test case updated in the ticket
Risks: low
Signed-off-by: vinod_kadungoth2@comcast.com

pstroffolino committed:
VPLAY-11265 spelling fixes (#570)

Reason for Change: fix few typos

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

pstroffolino committed:
VPLAY-11317 inband cc usage clarifications (#558)

VPLAY-11317 inband cc usage clarifications

Reason for Change: UVE Documentation improvements
Test Guidance: joint review with SMEs
Risk: None

Signed-off-by: Philip Stroffolino <philip_stroffolino@cable.comcast.com>

Vinish100 committed:
VPLAY-11352: Crash Observed while Playback of linear LLD HD/UHD (#563)

VPLAY-11352: Crash Observed while Playback of linear LLD HD/UHD (#563)
Reason for change: avoid double-free - mLldLowBuffObject is a child of telemetryParam, so it's automatically deleted.
Test Procedure: As mentioned in ticket
Risks: None

Signed-off-by: Vinish100 <vinish.balan@gmail.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/rdkcentral/aamp?pullRequest=584) <br/>All committers have signed the CLA.

psiva01 committed:
Merge branch 'develop' into feature/RDK-59589_federated_2510

paulpandian25 committed:
Reintroduced the condition check before calling clear() in clearDrmSession(), which was missed during the DRM refactoring (#619)

VPLAY-11530 [Charter XiOne | ES1][R38] Increase in PLTV XSTPP-999 Failures on both Xi1/ES1

Reason for Change : Reintroduced the condition check before calling clear() in clearDrmSession. This was lost during DRM refactoring.

Co-authored-by: vkadun208_comcast <vinod_kadungoth2@comcast.com>

psiva01 committed:
Revert "VPLAY-11140: Encrypted playback support with mp4demux (#536)"

This reverts commit 014b427fed2f3c8b5b14d58b50a85955496e8504.

psiva01 committed:
Merge pull request #624 from rdkcentral/feature/VPLAY-11140_Rvt_fed

Revert "VPLAY-11140: Encrypted playback support with mp4demux (#536)"

psiva01 committed:
Revert "Reintroduced the condition check before calling clear() in clearDrmSession(), which was missed during the DRM refactoring (#619)"

This reverts commit 62ab68d108d576f0fb3b6b447058657474fc2ca7.

psiva01 committed:
Reapply "Reintroduced the condition check before calling clear() in clearDrmSession(), which was missed during the DRM refactoring (#619)"

This reverts commit ab0a6468f4d4fa9884d1737c3aed5a569c426e7e.
